src/Tizen.Network.Stc/Tizen.Network.Stc/StcRule.cs

```diff
@@ -0,0 +1,373 @@
+                }
+                int ret = Interop.Stc.Rule.SetTimePeriod(_ruleHandle, value);
+                if (ret != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "Failed to set Time period, Error - " + (StcError)ret);
+                    StcErrorFactory.ThrowStcException(ret);
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the statistics information an application matched rule asynchronously.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="infoCb">The callback is called for each application that used network in between timestamps specified.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void GetStats(StatisticsCallback infoCb)
+        {
+            if (_disposed)
+            {
+                throw new ObjectDisposedException("Invalid StcRule instance (Object may have been disposed or released)");
+            }
+
+            _getStatsObjectCb = infoCb;
+
+            _getStatsCb = (int result, IntPtr info, IntPtr userData) =>
+            {
+                if(result != (int)StcError.None)
+                {
+                    Log.Error(Globals.LogTag, "GetStats failed, Error - " + (StcError)result);
+                    StcErrorFactory.ThrowStcException(result);
+                }
+                return _getStatsObjectCb(new NetworkStatistics(info));
+            };
+
+            int ret = Interop.Stc.Rule.GetStats(StcManagerImpl.Instance.GetSafeHandle(), _ruleHandle, _getStatsCb, IntPtr.Zero);
+            if (ret != (int)StcError.None)
+            {
+                Log.Error(Globals.LogTag, "GetStats() failed , Error - " + (StcError)ret);
+                StcErrorFactory.ThrowStcException(ret);
+            }
+        }
+
+        /// <summary>
+        /// Gets the statistics information of each application asynchronously.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <param name="infoCb">The callback is called for each application that used network in between timestamps specified.</param>
+        /// <privilege>http://tizen.org/privilege/network.get</privilege>
+        /// <exception cref="NotSupportedException">Thrown when the Stc is not supported.</exception>
+        /// <exception cref="UnauthorizedAccessException">Thrown when the permission is denied.</exception>
+        /// <exception cref="InvalidOperationException">Thrown when the method failed due to an invalid operation.</exception>
+        public void ForeachStats(StatisticsCallback infoCb)
```

Comment:
```suggestion
        public void Task<IEnumerable<NetworkStatistics>> GetAllStatistics()
```

It should be move to StcManager class

``` c#
 class StcManager {
...
  public static Task<NetworkStatistics> GetAllStatistics(StcRule rule)
...
```

Reply:
moved.